setup.py

```diff
@@ -13,18 +14,35 @@
 
 install_requires = open("requirements.txt").read().strip().split("\n")
 dev_requires = open("dev-requirements.txt").read().strip().split("\n")
+optional_requires = open("optional-requirements.txt").read().strip().split("\n")
+
+
+def optional_requirements(dependency_names: List[str]) -> List[str]:
+    requirements: List[str] = []
+
+    for dependency in dependency_names:
+        for optional_dependency in optional_requires:
+            if dependency in optional_dependency:
```

Comment:
this feels too blanket to me, we could get false-positives in some cases, such as
```
In [1]: "redis" in "fastapi-caching[redis]==1.0.0"       
Out[1]: True
```

maybe only have it match characters from the beginning 

```
In [5]: "redis" in "fastapi-caching[redis]==1.0.0"[:len( 
   ...: "redis")]
Out[5]: False

In [6]: "redis" in "redis==1.0.0"[:len("redis")]
Out[6]: True
```